src/ajax.js

```diff
@@ -462,14 +462,14 @@ jQuery.extend( {
 
 				// Raw string
 				getAllResponseHeaders: function() {
-					return state === 2 ? responseHeadersString : null;
+					return completed ? responseHeadersString : null;
 				},
 
 				// Caches the header
 				setRequestHeader: function( name, value ) {
-					var lname = name.toLowerCase();
-					if ( !state ) {
-						name = requestHeadersNames[ lname ] = requestHeadersNames[ lname ] || name;
+					if ( completed == null ) {
```

Comment:
I'm okay with that as long as we document it somewhere. The first thing I did when I saw this was look for a place where it might be assigned `null` thinking the check for both was important. The style guide example [implies](http://contribute.jquery.org/style-guide/js/#equality) we do it to check for both.